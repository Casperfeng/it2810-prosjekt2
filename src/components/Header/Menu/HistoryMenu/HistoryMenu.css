.historymenu {
  width: 140px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
}
.historyBtn {
  height: 60px;
  width: 45px;
}
