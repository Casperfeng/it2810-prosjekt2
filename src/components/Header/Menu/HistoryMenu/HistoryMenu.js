import React from 'react'
import './HistoryMenu.css';
class HistoryMenu extends React.Component { //TODO: Useless constructor?
  constructor(props) {
    super(props);
  }
  undoColor() {
    return this.props.historyIndex > 1 ? "white" : "#444444";
  }

  redoColor() {
    return this.props.historyIndex < this.props.historySize ? "white" : "#444444";
  }

  render() {
    return <div className="historymenu">
      <svg className="historyBtn" onClick = {this.props.handleUndo}>
        <polygon fill = {this.undoColor()} 
        points="25,15 25,22 41,22 41,38 25,38 25,45 5,30"/>
      </svg>
      <svg className="historyBtn" onClick = {this.props.handleRedo}>
        <polygon fill = {this.redoColor()}
        points="20,15 20,22 4,22 4,38 20,38 20,45 40,30"/>
      </svg>
    </div>
  }
}

export default HistoryMenu;