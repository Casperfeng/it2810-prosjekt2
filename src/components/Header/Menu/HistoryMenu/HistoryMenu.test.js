import React from 'react';
import HistoryMenu from './HistoryMenu';
import renderer from 'react-test-renderer';

it('renders correctly when unclicked', () => {
  const tree = renderer.create(<HistoryMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});
