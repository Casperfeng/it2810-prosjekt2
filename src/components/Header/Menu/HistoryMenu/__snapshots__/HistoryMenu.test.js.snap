// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when unclicked 1`] = `
<div
  className="historymenu"
>
  <svg
    className="historyBtn"
  >
    <polygon
      fill="#444444"
      points="25,15 25,22 41,22 41,38 25,38 25,45 5,30"
    />
  </svg>
  <svg
    className="historyBtn"
  >
    <polygon
      fill="#444444"
      points="20,15 20,22 4,22 4,38 20,38 20,45 40,30"
    />
  </svg>
</div>
`;
